collector/builder-config.yaml

```diff
@@ -106,11 +106,19 @@ connectors:
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector v0.106.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector v0.106.0
 
+providers:
+  - gomod: go.opentelemetry.io/collector/odigos/providers/odigosfileprovider v0.106.0 # fork default file provider for config reloading
+  - gomod: go.opentelemetry.io/collector/confmap/provider/envprovider v1.20.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpprovider v1.20.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpsprovider v1.20.0
```

Comment:
Not that I know of, so I included them to be safe. The reason is because right now, these are all silently included by default since we don't explicitly set any `providers` in our builder config.

Now that I'm adding the `odigosfileprovider` here, not providing these means they'll be removed from our build. If that's okay then for sure we can clean that up!

Reply:
I am pretty sure they are not used, so it's good opportunity to clean this unused component.
We can do it later on a different PR if you prefer